cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -0,0 +1,175 @@
+                if ("add".equals(op)) {
+                    // entry is not in the current, it is added
+                    updateOrAdd(pathValueWithoutSlash, desiredMap, updatedCE);
+                }
+            }
+
+            log.debug("Kafka Broker {} Logging Config Differs : {}", brokerId, d);
+            log.debug("Current Kafka Broker Logging Config path {} has value {}", pathValueWithoutSlash, lookupPath(source, pathValue));
+            log.debug("Desired Kafka Broker Logging Config path {} has value {}", pathValueWithoutSlash, lookupPath(target, pathValue));
+        }
+        return updatedCE;
+    }
+
+    private static String parseLogLevelFromAppenderCouple(String level) {
+        int index = level.indexOf(",");
+        if (index > 0) {
+            return level.substring(0, index).trim();
+        } else {
+            return level.trim();
+        }
+    }
+
+    private static void updateOrAdd(String propertyName, Map<String, String> desiredMap, Collection<AlterConfigOp> updatedCE) {
+        if (!propertyName.contains("log4j.appender") && !propertyName.equals("monitorInterval")) {
+            String level = parseLogLevelFromAppenderCouple(desiredMap.get(propertyName));
+            if (isValidLoggerLevel(level)) {
+                updatedCE.add(new AlterConfigOp(new ConfigEntry(propertyName, level), AlterConfigOp.OpType.SET));
+                log.trace("{} not set in current or has deprecated value. Setting to {}", propertyName, level);
+            } else {
+                log.warn("Level {} is not valid logging level", level);
+            }
+        }
+    }
+
+    /**
+     * All loggers can be set dynamically. If the logger is not set in desire, set it to ERROR. Loggers with already set to ERROR should be skipped.
+     * ERROR is set as inactive because log4j does not support OFF logger value.
+     * We want to skip "root" logger as well to avoid duplicated key in alterConfigOps collection.
+     * @param alterConfigOps collection of AlterConfigOp
+     * @param pathValueWithoutSlash name of "removed" logger
+     * @param entry entry to be removed (set to ERROR)
+     */
+    private static void removeProperty(Collection<AlterConfigOp> alterConfigOps, String pathValueWithoutSlash, ConfigEntry entry) {
+        if (!pathValueWithoutSlash.contains("log4j.appender") && !pathValueWithoutSlash.equals("root") && !"ERROR".equals(entry.value())) {
+            alterConfigOps.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, "ERROR"), AlterConfigOp.OpType.SET));
+            log.trace("{} not set in desired, setting to ERROR", entry.name());
+        }
+    }
+
+    /**
+     * @return whether the current config and the desired config are identical (thus, no update is necessary).
+     */
+    @Override
+    public boolean isEmpty() {
+        return  diff.size() == 0;
+    }
+
+    private static boolean isValidLoggerLevel(String level) {
+        return VALID_LOGGER_LEVELS.contains(level);
```

Comment:
Why did use use a `List` for `VALID_LOGGER_LEVELS` if you're only needing `contains()`? A `HashSet` will be more efficient since it likely needs only 1 `hash()` and 1 or maybe 2 `equals()`.